app/prepare_proposal.go

```diff
@@ -32,7 +32,23 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 		ante.DefaultSigVerificationGasConsumer,
 		app.IBCKeeper,
 	)
-	txs := FilterTxs(sdkCtx, handler, app.txConfig, req.BlockData.Txs)
+
+	var txs [][]byte
+	if app.LastBlockHeight() == 0 {
```

Comment:
[optional]

I'll defer to the author if its needed or not, but it might be nice to briefly explain this or link to an issue that does for more context

Reply:
Yeah since this is somewhat of an unusual hack I think it would be good to point out what why this is needed

Reply:
Sure, please see the added explanations https://github.com/celestiaorg/celestia-app/pull/2233/commits/93692c5d4876e18d3c4f366e8d9126df608679c4.